src/Tools/IdeBenchmarks/SQLitePersistentStorageBenchmark.cs

```diff
@@ -0,0 +1,143 @@
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.SolutionSize;
+using Microsoft.CodeAnalysis.SQLite;
+using Microsoft.CodeAnalysis.Storage;
+using Microsoft.CodeAnalysis.Test.Utilities;
+
+namespace IdeBenchmarks
+{
+    public class SQLitePersistentStorageBenchmarks
+    {
+        private readonly UseExportProviderAttribute _useExportProviderAttribute = new UseExportProviderAttribute();
+
+        // Run the test with different ratios of reads/writes.
+        [Params(0, 25, 50, 75, 100)]
+        public int ReadPercentage { get; set; }
+
+        private TestWorkspace _workspace;
+        private SQLitePersistentStorageService _storageService;
+        private IChecksummedPersistentStorage _storage;
+        private Document _document;
+        private Random _random;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            _useExportProviderAttribute.Before(null);
+
+            if (_workspace != null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            _workspace = TestWorkspace.Create(
+@"<Workspace>
+    <Project Language=""NoCompilation"" CommonReferences=""false"">
+        <Document>
+            // a no-compilation document
+        </Document>
+    </Project>
+</Workspace>");
+
+            // Ensure we always use the storage service, no matter what the size of the solution.
+            _workspace.Options = _workspace.Options.WithChangedOption(StorageOptions.SolutionSizeThreshold, -1);
+
+            _storageService = new SQLitePersistentStorageService(
+                _workspace.Services.GetService<IOptionService>(),
+                new LocationService(),
+                new SolutionSizeTracker());
+
+            _storage = _storageService.GetStorageWorker(_workspace.CurrentSolution);
+            if (_storage == NoOpPersistentStorage.Instance)
+            {
+                throw new InvalidOperationException("We didn't properly get the sqlite storage instance.");
+            }
+
+            Console.WriteLine("Storage type: " + _storage.GetType());
+            _document = _workspace.CurrentSolution.Projects.Single().Documents.Single();
+            _random = new Random(0);
+        }
+
+        [GlobalCleanup]
+        public void GlobalCleanup()
+        {
+            if (_workspace == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            _document = null;
+            _storage.Dispose();
+            _storage = null;
+            _storageService = null;
+            _workspace.Dispose();
+            _workspace = null;
+
+            _useExportProviderAttribute.After(null);
+        }
+
+        private static readonly byte[] s_bytes = new byte[1000];
+
+        [Benchmark(Baseline = true)]
+        public Task Perf()
+        {
+            var tasks = new List<Task>();
```

Comment:
Can avoid a few unnecessary array allocations as the list grows.

```suggestion
            var tasks = new List<Task>(1000);
```

Reply:
Enh :)

Reply:
This should probably get fixed since it's inside the benchmark measurement loop so it impacts the results.